includes/Commerce/Orders.php

```diff
@@ -82,10 +82,86 @@ public function create_local_order( Order $remote_order ) {
 	 * @param Order $remote_order Orders API order object
 	 * @param \WC_Order $local_order local order object
 	 * @return \WC_Order
+	 * @throws SV_WC_Plugin_Exception|\WC_Data_Exception
 	 */
 	public function update_local_order( Order $remote_order, \WC_Order $local_order ) {
 
-		// TODO: implement
+		// add/update items
+		foreach ( $remote_order->get_items() as $item ) {
+
+			$wc_product_id = $item['retailer_id'];
+
+			$matching_wc_order_item = false;
+
+			// check if the local order already has this item
+			foreach ( $local_order->get_items() as $wc_order_item ) {
+
+				if ( $wc_order_item instanceof \WC_Order_Item_Product && $wc_product_id === (string) $wc_order_item->get_product_id() ) {
+					$matching_wc_order_item = $wc_order_item;
+					break;
+				}
+			}
+
+			if ( empty( $matching_wc_order_item ) ) {
+
+				$wc_product = wc_get_product( $wc_product_id );
+
+				if ( ! $wc_product instanceof \WC_Product ) {
+					throw new SV_WC_Plugin_Exception( "Product with WC ID $wc_product_id not found" );
+				}
+
+				$matching_wc_order_item = new \WC_Order_Item_Product();
+				$local_order->add_item( $matching_wc_order_item );
+			}
+
+			$matching_wc_order_item->set_product_id( $wc_product_id );
+			$matching_wc_order_item->set_quantity( $item['quantity'] );
+			$matching_wc_order_item->set_subtotal( $item['quantity'] * $item['price_per_unit']['amount'] );
+			// TODO: should we use the estimated_tax or the captured_tax on the line below?
+			$matching_wc_order_item->set_subtotal_tax( $item['tax_details']['estimated_tax']['amount'] );
+			$matching_wc_order_item->save();
+		}
+
+		// update information from selected_shipping_option
+		$selected_shipping_option = $remote_order->get_selected_shipping_option();
+
+		$local_order->set_shipping_total( $selected_shipping_option['price']['amount'] );
+		$local_order->set_shipping_tax( $selected_shipping_option['calculated_tax']['amount'] );
+
+		// update information from shipping_address
+		$shipping_address = $remote_order->get_shipping_address();
```

Comment:
Let's code defensively here and only set this customer data if it's returned in the API order. iirc I thought I remember seeing some case where pending orders don't send along customer data until it's finalized.

Reply:
Does it return all the fields as empty strings or an empty array?

Reply:
I don't know - the docs aren't clear on that so it'll need to be tested against a real API response as well

Reply:
Updated in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1515/commits/abe1ac3dbb93d742492913328bd58b8e4cfceac0